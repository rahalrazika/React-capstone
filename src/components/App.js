import '../style/App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import NavBar from './Navbar';
import ProductsList from './ProductsList';
import DetailPage from './DetailPage';

function App() {
  return (
    <div className="App">
      <Router>
        <NavBar />
        <Switch>
          <Route path="/" exact component={ProductsList} />
          <Route path="/product/:productId" component={DetailPage} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
