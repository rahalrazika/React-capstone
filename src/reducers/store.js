import { createStore } from 'redux';
import rootReducer from './index';

const store = createStore(rootReducer);
export default store;
