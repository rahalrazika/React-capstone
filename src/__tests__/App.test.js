import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../components/App';
import store from '../reducers/store';

it('Renders DetailPage component', () => {
  const page = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  ).toJSON();
  expect(page).toMatchSnapshot();
});
