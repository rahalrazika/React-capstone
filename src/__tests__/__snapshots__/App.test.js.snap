// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders DetailPage component 1`] = `
<div
  className="App"
>
  <nav
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="navbar-logo"
        src="https://static.vecteezy.com/system/resources/previews/002/114/693/original/online-shop-logo-template-design-illustration-vector.jpg"
      />
    </a>
    <div
      className="nav-search"
    >
      <input
        className="search-input"
        onChange={[Function]}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root nav-search-icon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      className="nav-links-container"
    >
      <a
        className="nav-links"
        href="/"
        onClick={[Function]}
      >
        <div
          className="nav-options"
        >
          <span
            className="text-one"
          >
            Return
          </span>
          <span
            className="text-two"
          >
            Home
          </span>
        </div>
      </a>
    </div>
  </nav>
  <div
    className="productContainer"
  >
    <div
      className="banner"
    >
      <img
        alt="banner"
        src="banner.jpg"
      />
    </div>
    <div
      className="container d-flex"
    >
      <div
        className="lds-spinner"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
